12-watchtower-API.md

```diff
@@ -127,122 +129,167 @@ This message contains all the information regarding the appointment that the cli
+* MUST set `transaction_size` to the size of the serialized `justice_transaction`, in bytes.
+* MUST set `transaction_fee` to the fee set in the `justice_transaction`.
+* MUST set `cipher` to the cipher used to create the `encrypted_blob`.
 * if `qos` was agreed on `wt_init`:
 	* MUST set the `op_customer_signature_algorithm` to one of the signature algorithms agreed on `wt_init`.
 	* MUST set `op_customer_signature` to the signature of the appointment using `op_customer_signature_algorithm`.
-	* MUST set `op_customer_public_key` the public key that matches the private key used to create `op_customer_signature`.
+	* MUST set `op_customer_public_key` to the public key that matches the private key used to create `op_customer_signature`.
 
 The receiving node:
+* upon receiving a `transaction_fee`:
+	* MUST compute the `fee_rate` set in the `justice_tx`.
 
-* upon receiving a `locator` that is not a `16-byte` value:
-	* MUST reject the appointment.
-* upon receiving a `start_block` behind the current chain tip:
-	* MUST reject the appointment.
-* upon receiving a `start_block` too close to the current chain tip:
-	* SHOULD reject the appointment.
-* upon receiving a `start_block` too far away in the future:
-	* SHOULD reject the appointment.
-* upon receiving an `end_block` behind the current chain tip:
-	* MUST reject the appointment.
-* upon receiving an `end_block` too far away in the future:
-	* SHOULD reject the appointment.
-* upon receiving a `dispute_delta` too small:
-	* SHOULD reject the appointment.
-* upon receiving an `encrypted_blob` of non-feasible size:
-	* MUST reject the appointment.
-* upon receiving a `cipher` that he does not implement:
-	* MUST reject the appointment. 
-* upon receiving a `transaction_size` too big:
-	* MUST reject the appointment.
-* upon receiving a `transaction_size` too small:
-	* MUST reject the appointment.
-* upon receiving a `transaction_fee` too low:
-	* SHOULD reject the appointment.
+The receiving node MUST reject the appointment if:
 
-* if `qos` was agreed on `wt_init`:
-	* if `op_customer_signature_algorithm` is missing:
-		* MUST reject the appointment.
-	* if `op_customer_signature_algorithm` does not match with one of the supported signing algorithms:
-		* MUST reject the appointment.
-	* if `op_customer_signature` is missing:
-		* MUST reject the appointment.
-	* if `op_customer_public_key` is missing:
-		* MUST reject the appopintment.
-	* if `op_customer_signature` cannot be verified using `op_customer_public_key`:
-		* MUST reject the appopintment.
-
-* if `qos` was not agreed on `wt_init`:
-	* if `op_customer_signature_algorithm` is present:
-		* SHOULD reject the appointment.
-	* if `op_customer_signature` is present:
-		* SHOULD reject the appointment.
-	* if `op_customer_public_key` is present:
-		* SHOULD reject the appopintment.
+* `locator` is not a `16-byte` value.
+* `start_block` is not an integer.
+* `start_block` is behind the current chain tip.
```

Comment:
Then you may have missed the trigger already since the block has already been processed. This is not a big deal honestly, we could change it to not behind the tip and allow the tip being also valid.

Reply:
it would just allow it from tip, and then you would immediately check if you need to go for it. 

i think the issue is more, it shouldnt be "too far in the past" i.e. it should be within 6 blocks of the tip.

Reply:
actually if this is reject reason.

then the start block should not be too far in the past or too far in the future (even, not greater than the current tip).

Reply:
I'll do a -1/+1 window (so past, current, and future block). That should cover out-of-sync cases. For now I'm gonna leave it like that and therefore prevent hiring for future jobs (makes the protocol easier and prevent having to store arbitrary long starting appointments)

Reply:
ok